### ISSUE TITLE
SC1004 false positive?

### ISSUE THREAD

**Issue by dylanaraps** [koalaman/shellcheck#1246]
The script has a line length limit of `100` chars. This means that longer `awk` commands need to be split on multiple lines. Using a backslash causes shellcheck to flag it as an error. Is there a better way of splitting these commands on new lines?

#### For bugs
- Rule Id (if any, e.g. SC1000): SC1004
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
#
# Snippet from script. It doesn't work on it's own but it's an example for shellcheck. 
# (Shellcheck will still complain.)
[[ -z "$wm_theme" ]] && \
    wm_theme="$(awk '/library=org.kde/{gsub(/library=org.kde./,"",$0);\
                     print $0; exit}' "$kwinrc")"
[[ -z "$wm_theme" ]] && \
    wm_theme="$(awk '/PluginLib=kwin3_/{gsub(/PluginLib=kwin3_/,"",$0);\
                     print $0; exit}' "$kwinrc")"
```

#### Here's what shellcheck currently says:

```
SC1004: This backslash+linefeed is literal. Break outside single quotes if you just want to break the line.
```

#### Here's what I wanted or expected to see:

No error.

**Comment by koalaman** [comment]
It's rare for commands to be ok with backslash+linefeed in their arguments, but `awk` does allow that. However, it does just as well with regular linefeeds, so you can just use:

    wm_theme="$(awk '/PluginLib=kwin3_/ {
                        gsub(/PluginLib=kwin3_/, "", $0);
                        print $0;
                        exit
                     }' "$kwinrc")"

**Comment by jetzerb** [comment]
But `sed` is pickier than `awk`, and _requires_ the backslash, as illustrated with this script:

```sh
#!/usr/bin/env bash

#
# Demonstrate False Positive SC1004 via sed
#


# sed executes successfully, but generates an SC1004
echo "sed with backslash:"
echo "foo" | sed 's/$/\
bar/;'

# Same as above, but without the trailing slash
# sed throws an "unterminated subsitution" error, but it passes shellcheck
echo "sed without backslash:"
echo "foo" | sed 's/$/
bar/;'
```

Results:
```
$ shellcheck sc1004_test.sh 

In sc1004_test.sh line 10:
echo "foo" | sed 's/$/\
                      ^-- SC1004: This backslash+linefeed is literal. Break outside single quotes if you just want to break the line.

For more information:
  https://www.shellcheck.net/wiki/SC1004 -- This backslash+linefeed is litera...

$ sh sc1004_test.sh 
sed with backslash:
foo
bar
sed without backslash:
sed: -e expression #1, char 4: unterminated `s' command
```

**Comment by jessehu** [comment]
Here is the workaround:

```
  # shellcheck disable=SC1004
  sed -i -e '3i\
---
' "$1"
```
